plugin/evm/orderbook/memory_database.go

```diff
@@ -1076,16 +1094,26 @@ func (db *InMemoryDatabase) determineOrdersToCancel(addr common.Address, trader
 	foundCancellableOrders := false
 	for _, order := range traderOrders {
 		// @todo how are reduce only orders that are not fillable cancelled?
-		if order.ReduceOnly || order.OrderType != Limit {
+		if order.OrderType != Limit {
+			// ioc and signed orders dont need to be cancelled
 			continue
 		}
-		ordersToCancel[addr] = append(ordersToCancel[addr], order)
-		foundCancellableOrders = true
-		orderNotional := big.NewInt(0).Abs(hu.Div1e18(hu.Mul(order.GetUnFilledBaseAssetQuantity(), order.Price))) // | size * current price |
-		marginReleased := hu.Div1e6(hu.Mul(orderNotional, db.configService.getMinAllowableMargin()))
-		_availableMargin.Add(_availableMargin, marginReleased)
-		if _availableMargin.Sign() >= 0 {
-			break
+		if order.ReduceOnly {
+			if marketsToCancelReduceOnlyOrdersIn[order.Market] {
+				ordersToCancel[addr] = append(ordersToCancel[addr], order)
+				marketsToCancelReduceOnlyOrdersIn[order.Market] = false // keeping it simple by cancelling just 1 stale order per market per run
+			} else {
+				continue
```

Comment:
this bit is not required, can be removed

Reply:
done